include/graphblas/bsp1d/io.hpp

```diff
@@ -823,6 +823,49 @@ namespace grb {
 		return ret;
 	}
 
+	template<
+		Descriptor descr = descriptors::no_operation,
+		typename DataType, typename RIT, typename CIT, typename NIT,
+		typename ValueType = DataType
+	>
+	RC set(
+		Matrix< DataType, BSP1D, RIT, CIT, NIT  > &C,
+		const ValueType& val,
+		const Phase &phase = EXECUTE
+	) noexcept {
+		const size_t m = nrows( C );
+		const size_t n = ncols( C );
+		const size_t nz = nnz( C );
+
```

Comment:
dynamic error checks seem missing: for example, a guard against overflow if m * n > 2^64-1. I think in such a case ILLEGAL should be returned, which in turn then also modifies the base specification. This may play for more primitives actually and we should open up an issue for that